dm/dm/config/subtask.go

```diff
@@ -274,8 +274,9 @@ type SubTaskConfig struct {
 	} `yaml:"experimental" toml:"experimental" json:"experimental"`
 
 	// below member are injected by dataflow engine
-	ExtStorage     extstorage.ExternalStorage `toml:"-" json:"-"`
-	MetricsFactory promutil.Factory           `toml:"-" json:"-"`
+	ExtStorage      extstorage.ExternalStorage `toml:"-" json:"-"`
+	MetricsFactory  promutil.Factory           `toml:"-" json:"-"`
+	FrameworkLogger *zap.Logger                `toml:"-" json:"-"`
```

Comment:
Can we pass the logger by NewDumpling instead of add it in config?

Reply:
the current way is: engine inject logger to subtask config, worker call NewDumpling(subtask).

You proposed way is: engine inject logger to subtask config, worker call NewDumpling(subtask, subtask.FrameworkLogger)? that's no gain.

Reply:
I mean worker call NewDumpling(subtask, logger), logger no need to inject to config.

Reply:
I see, we can let engine worker's unit holder call NewDumpling, NewLightning, NewSyncer ... will take a look later